File: src/main/java/hilfling/backend/hilfing/model/AnalogPhoto.java
package hilfling.backend.hilfing.model;

import lombok.Data;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Table;

@Data
@Entity
@Table(name = "analog_photo")
public class AnalogPhoto extends Photo {

    public AnalogPhoto(
            String smallUrl,
            String mediumUrl,
            String largeUrl,
            Boolean goodPicture,
            Motive motive,
            Integer imageNumber,
            Integer pageNumber
    ) {
        super(smallUrl, mediumUrl, largeUrl, goodPicture, motive);
        this.imageNumber = imageNumber;
        this.pageNumber = pageNumber;
    }

    @Column(name = "image_number", nullable = false)
    private Integer imageNumber;

    @Column(name = "page_number", nullable = false)
    private Integer pageNumber;
}

File: src/main/java/hilfling/backend/hilfing/model/AnalogPhoto.kt
package hilfling.backend.hilfing.model

import javax.persistence.*

@Entity
@Table(name = "analog_photo")
class AnalogPhoto(
        smallUrl: String = "",
        mediumUrl: String = "",
        largeUrl: String = "",
        goodPicture: Boolean = false,
        motive: Motive,
        place: Place,
        securityLevel: SecurityLevel,
        gang: Gang,
        photoGangBanger: PhotoGangBanger,
        @Column(name = "image_number", nullable = false)
        val imageNumber: Int,
        @Column(name = "page_number", nullable = false)
        val pageNumber: Int
): Photo(
        smallUrl,
        mediumUrl,
        largeUrl,
        goodPicture,
        motive,
        place,
        securityLevel,
        gang,
        photoGangBanger
)